sdk/nodejs/serverless/function.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import * as pulumi from "@pulumi/pulumi";
+import * as cloudfunctions from "../cloudfunctions";
+import * as storage from "../storage";
+
+import * as filepath from "path";
+import * as readPackageJson from "read-package-json";
+
+/**
+ * Handler is the signature for a serverless function.
+ */
+export type Handler = (request: any, response: any) => any;
```

Comment:
Let's go ahead and use the types from `express` since Google Cloud does really provide live instances of those types - so really handy to get the rich typing automatically here.  See https://github.com/pulumi/examples/blob/master/gcp-ts-functions/gcpFunction.ts#L11.